packages/runtime-core/src/componentProps.ts

```diff
@@ -274,6 +274,7 @@ function setFullProps(
   attrs: Data
 ) {
   const [options, needCastKeys] = instance.propsOptions
+  const rawCurrentProps = extend({}, toRaw(props))
   if (rawProps) {
     for (const key in rawProps) {
       const value = rawProps[key]
```

Comment:
I think this will be better
```javascript
// key, ref are reserved and never passed down
if (isReservedProp(key)) {
   continue
}
const value = rawProps[key]
```